tests/unit/test_rbac_resolvers_key_value_pair.py

```diff
@@ -17,39 +17,310 @@
+from st2common.persistence.auth import User
+from st2common.persistence.rbac import Role
+from st2common.persistence.rbac import UserRoleAssignment
+from st2common.persistence.rbac import PermissionGrant
 from st2common.persistence.keyvalue import KeyValuePair
 from st2common.models.db.keyvalue import KeyValuePairDB
+from st2common.models.db.auth import UserDB
+from st2common.models.db.rbac import RoleDB
+from st2common.models.db.rbac import UserRoleAssignmentDB
+from st2common.models.db.rbac import PermissionGrantDB
 
 from st2rbac_backend.resolvers import KeyValuePermissionsResolver
 from tests.unit.test_rbac_resolvers import BasePermissionsResolverTestCase
 
-__all__ = [
-    'KeyValuePermissionsResolver'
-]
+__all__ = ["KeyValuePermissionsResolver"]
 
 
 class KeyValuePermissionsResolverTestCase(BasePermissionsResolverTestCase):
     def setUp(self):
         super(KeyValuePermissionsResolverTestCase, self).setUp()
 
-        kvp_1_db = KeyValuePairDB(name='key1', value='val1')
+        user_1_db = UserDB(name="1_role_view_grant")
+        user_1_db = User.add_or_update(user_1_db)
+        self.users["custom_role_key_value_pair_view_grant"] = user_1_db
+
+        user_2_db = UserDB(name="2_role_list_grant")
+        user_2_db = User.add_or_update(user_2_db)
+        self.users["custom_role_key_value_pair_list_grant"] = user_2_db
+
+        user_3_db = UserDB(name="3_role_all_grant")
+        user_3_db = User.add_or_update(user_3_db)
+        self.users["custom_role_key_value_pair_all_grant"] = user_3_db
+
+        user_4_db = UserDB(name="4_role_create_grant")
+        user_4_db = User.add_or_update(user_4_db)
+        self.users["custom_role_key_value_pair_create_grant"] = user_4_db
+
+        user_5_db = UserDB(name="5_role_delete_grant")
+        user_5_db = User.add_or_update(user_5_db)
+        self.users["custom_role_key_value_pair_delete_grant"] = user_5_db
+
+        kvp_1_db = KeyValuePairDB(name="key1", value="val1")
         kvp_1_db = KeyValuePair.add_or_update(kvp_1_db)
-        self.resources['kvp_1'] = kvp_1_db
+        self.resources["user_role_1"] = kvp_1_db
+
+        kvp_2_db = KeyValuePairDB(name="key2", value="val2")
+        kvp_2_db = KeyValuePair.add_or_update(kvp_2_db)
+        self.resources["user_role_2"] = kvp_2_db
+
+        kvp_3_db = KeyValuePairDB(name="key3", value="val3")
+        kvp_3_db = KeyValuePair.add_or_update(kvp_3_db)
+        self.resources["user_role_3"] = kvp_3_db
+
+        kvp_4_db = KeyValuePairDB(name="key4", value="val4")
+        kvp_4_db = KeyValuePair.add_or_update(kvp_4_db)
+        self.resources["user_role_4"] = kvp_4_db
+
+        kvp_5_db = KeyValuePairDB(name="key5", value="val5")
+        kvp_5_db = KeyValuePair.add_or_update(kvp_5_db)
+        self.resources["user_role_5"] = kvp_5_db
+
+        kvp_6_db = KeyValuePairDB(name="key6", value="val6")
+        kvp_6_db = KeyValuePair.add_or_update(kvp_6_db)
+        self.resources["user_role_6"] = kvp_6_db
```

Comment:
Are these unit tests for system scope or user scope key value pairs? Do we have unit tests for both? It is not obvious here.